dipy/reconst/sfm.py

```diff
@@ -489,22 +580,28 @@ def fit(self, data, mask=None):
 
 
 class SparseFascicleFit(ReconstFit):
+
     def __init__(self, model, beta, S0, iso):
         """
         Initalize a SparseFascicleFit class instance
 
         Parameters
         ----------
         model : a SparseFascicleModel object.
+
         beta : ndarray
             The parameters of fit to data.
+
         S0 : ndarray
             The mean non-diffusion-weighted signal.
+
         iso : IsotropicFit class instance
             A representation of the isotropic signal, together with parameters
             of the isotropic signal in each voxel, that is capable of
             deriving/predicting an isotropic signal, based on a gradient-table.
         """
+        super().__init__(self, model)
```

Comment:
I believe that if you add this, you can remove the assignment `self.model` below on L605.

Reply:
If I do that, then the following occurs:
```
    def odf(self, sphere):
        """
        The orientation distribution function of the SFM
    
        Parameters
        ----------
        sphere : Sphere
            The points in which the ODF is evaluated
    
        Returns
        -------
        odf :  ndarray of shape (x, y, z, sphere.vertices.shape[0])
    
        """
>       odf_matrix = self.model.cache_get('odf_matrix', key=sphere)
E       AttributeError: 'SparseFascicleFit' object has no attribute 'cache_get'
```
for the subsequent `odf` method.

Reply:
OK - I'm probably wrong then!